server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -4997,6 +4997,7 @@ public PhysicalNetworkTrafficType addTrafficTypeToPhysicalNetwork(Long physicalN
             if (TrafficType.Public.equals(trafficType)) {
                 List<String> isolationMethods = network.getIsolationMethods();
                 if ((isolationMethods.size() == 1 && isolationMethods.get(0).toLowerCase().equals("vxlan"))
+                        // TODO: what happens if there are multiple isolation methods, why are multiple isolation methods allowed per pnet? They seem to be filterable by provider/preifx (PhysicalNetwork.java). Documentation says only two are allowed but code and tests contradict this.
```

Comment:
I see [checks](https://github.com/apache/cloudstack/blob/main/server/src/main/java/com/cloud/network/NetworkServiceImpl.java#L3824-L3824) to ensure only one isolation method can exist at this time. @rohityadavcloud @DaanHoogland I would like to confirm if the API field should be plural for futureproofing, or should we maintain parity with with [addTrafficType](https://cloudstack.apache.org/api/apidocs-4.17/apis/addTrafficType.html), and use singular?

Reply:
Yes, that would be okay.

Reply:
> Yes, that would be okay.

to summarize - even though string array is getting serialized correctly, I will stick to how it is handled currently in the [PhysicalNetworkResponse](https://github.com/apache/cloudstack/blob/a06f8a8763dc0d30b14e41d579ec73d40613e1d9/api/src/main/java/org/apache/cloudstack/api/response/PhysicalNetworkResponse.java#L129) where the methods array is converted into a string csv value.

Reply:
comma-separated string (vlans) is completed fine.